lib/client/client.js

```diff
@@ -236,53 +222,83 @@ function checkRequestResponse(self, response, result, callback) {
   return callback(null, res);
 }
 
-function requestCallback(self, err, response, result, callback) {
-  if (err) {
-    return callback(err);
-  }
+function _assembleUrl(self, uri) {
+  var lastElement,
+      params = '';
 
-  checkRequestResponse(self, response, result, function(err, res) {
-    if (err) {
-      return callback(err);
-    }
-    var body = null;
-    if (res) {
-      if (self.jsonAPINames){
-        for (var i = 0; i < self.jsonAPINames.length; i++){
-          if (res.hasOwnProperty(self.jsonAPINames[i].toString())){
-            body = res[self.jsonAPINames[i].toString()];
-            break;
-          }
+  if ('object' === typeof uri && Array.isArray(uri)) {
+    lastElement = uri.pop();
+    if (lastElement) {
+      // we have received an object ex. {"sort_by":"id","sort_order":"desc"}
+      if ('object' === typeof lastElement) {
+        if (self.sideLoad.length){
+          lastElement.include = self.sideLoad.join(',');
         }
+        params = '?' + qs.stringify(lastElement);
       }
-
-      if (!body) {
-        body = res;
+      // we have received a query string ex. '?sort_by=id&sort_order=desc'
+      else if (0 === lastElement.toString().indexOf('?')) {
+        if (self.sideLoad.length){
+          lastElement += '&include='+ self.sideLoad.join(',');
+        }
+        params = lastElement;
       }
-      if (self.hasOwnProperty('sideLoadMap')){
-        body = populateFields(body, res, self.sideLoadMap);
+      else {
+        if (self.sideLoad.length){
+          params = '?include=' + self.sideLoad.join(',');
+        }
+        uri.push(lastElement);
       }
-    } else {
-      body = '';
     }
+    return self.options.get('remoteUri') + '/' + uri.join('/') + '.json' + params;
+  }
+  else if ('string' === typeof uri && uri.indexOf(self.options.get('remoteUri')) === -1) {
+    return self.options.get('remoteUri') + uri;
+  }
+  else {
+    return uri;
+  }
+}
 
-    return callback(null, response.statusCode, body, response, res);
-  });
+function _auth (self) {
+  let auth = self.options.get('password') ? ':' + self.options.get('password') : '/token:' + self.options.get('token');
+  let encoded  = new Buffer(self.options.get('username') + auth).toString('base64');
+  let useOAuth = self.options.get('oauth');
+  let token    = self.options.get('token');
+  return typeof useOauth !== 'undefined' ? 'Bearer ' + token : 'Basic ' + encoded
+}
 
+function _findAllRecordsById(data, key, id){
+  var arr = [];
+  for (var i = 0; i < data.length; i++){
+    if (data[i][key] === id){
+      arr.push(data[i]);
+    }
+  }
+  return arr;
 }
 
-function populateFields(data, response, map){
+function _findOneRecordById(data, key, id){
+  for (var i = 0; i < data.length; i++){
+    if (data[i][key] === id){
+      return data[i];
+    }
+  }
+  return null;
```

Comment:
For simplicity you could probably use
```js
return data.find(function (item) {
  return item[key] === id
})
```
Or if you do have ES6 support
```js
return data.find(item => item[key] === id)
```
And instead of `_findAllRecordsById` use the same as above but with `data.filter(...)` instead of find